Source/Scene/Scene.js

```diff
@@ -3497,20 +3531,24 @@ define([
             limit = Number.MAX_VALUE;
         }
 
-        var pickedResult = this.pick(windowPosition, width, height);
-        while (defined(pickedResult) && defined(pickedResult.primitive)) {
-            result.push(pickedResult);
-            if (0 >= --limit) {
-                break;
+        var pickedResult = pickCallback();
+        while (defined(pickedResult) && defined(pickedResult.object) && defined(pickedResult.object.primitive)) {
+            var object = pickedResult.object;
+            var primitive = object.primitive;
+
+            if (!isExcluded(object, objectsToExclude)) {
```

Comment:
The way I thought we would have implemented this is:
```
foreach excluded - save current show, set show to false
drill pick as usual
foreach excluded - restore show
```
Is the current approach cleaner?  Seems slower since it could require extra scene renders if I am following it correctly.

Reply:
That's the approach I was hoping to do but an entity's show only gets registered after the next clock tick. I had opened https://github.com/AnalyticalGraphicsInc/cesium/issues/7024 to address this but never looked closer into changing the update flow.

Reply:
What is involved to fix #7024?  Seems like the common case, e.g., for a 3D model, will be that the scene is rendered twice, which feels like an unacceptable performance hit.  Maybe backface culling will help for clean models, but I would not rely on it.

What other options are there to do this in a single render for the common case?